jdaviz/configs/mosviz/tests/test_parsers.py

```diff
@@ -6,6 +7,75 @@
 from jdaviz.utils import PRIHDR_KEY, COMMENTCARD_KEY
 
 
+@pytest.mark.filterwarnings('ignore')
+@pytest.mark.remote_data
+@pytest.mark.parametrize('instrument_arg', ('nirspec', None))
+def test_nirspec_parser(mosviz_helper, tmpdir, instrument_arg):
+    '''
+    Tests loading our default MosvizExample notebook data
+    Also tests no instrument keyword fallback, and IntraRow linking
+    '''
+
+    test_data = 'https://stsci.box.com/shared/static/ovyxi5eund92yoadvv01mynwt8t5n7jv.zip'
+    fn = download_file(test_data, cache=True, timeout=30)
+    with ZipFile(fn, 'r') as sample_data_zip:
+        sample_data_zip.extractall(tmpdir)
+
+    level3_path = (pathlib.Path(tmpdir) / 'mosviz_nirspec_data_0.3' / 'level3')
+
+    data_dir = level3_path
+
+    if instrument_arg:
+        mosviz_helper.load_data(directory=data_dir, instrument=instrument_arg)
+    else:
+        # When no instrument is provided, mosviz.load_data should fallback to the nirspec loader.
+        # Naturally, the nirspec dataset should then work without any instrument keyword
+        with pytest.warns(UserWarning, match="Ambiguous MOS Instrument"):
+            mosviz_helper.load_data(directory=data_dir)
+
+    assert len(mosviz_helper.app.data_collection) == 16
+
+    # MOS Table meta should be empty:
+    assert len(mosviz_helper.app.data_collection["MOS Table"].meta) == 0
+
+    # Check that the data was loaded in the same order we expect:
+    assert mosviz_helper.app.data_collection[15].meta['SOURCEID'] == 2315
+    for i in range(0, 5):
+        assert mosviz_helper.app.data_collection[i+1].label == f"Image {i}"
+
+        spec1d = mosviz_helper.app.data_collection[i+6]
```

Comment:
Would be nice to explain the magic numbers (5, 6, 11) so that if someone changes the ZIP file content in the future, we know how to update this.

Reply:
Great point! Added comments here: https://github.com/spacetelescope/jdaviz/pull/1836/commits/361571461baafbd888a69a367a99af9cc028435b